runtime/ms-rest-azure/lib/login.js

```diff
@@ -468,4 +470,232 @@ exports.withServicePrincipalSecretWithAuthResponse = function withServicePrincip
+  }
+  if (!credsObj.clientId) {
+    throw new Error(`"clientId" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.clientSecret) {
+    throw new Error(`"clientSecret" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.subscriptionId) {
+    throw new Error(`"subscriptionId" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.tenantId) {
+    throw new Error(`"tenantId" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.activeDirectoryEndpointUrl) {
+    throw new Error(`"activeDirectoryEndpointUrl" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.resourceManagerEndpointUrl) {
+    throw new Error(`"resourceManagerEndpointUrl" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.activeDirectoryGraphResourceId) {
+    throw new Error(`"activeDirectoryGraphResourceId" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.sqlManagementEndpointUrl) {
+    throw new Error(`"sqlManagementEndpointUrl" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.galleryEndpointUrl) {
+    throw new Error(`"galleryEndpointUrl" is missing from the auth file: ${filePath}.`);
+  }
+  if (!credsObj.managementEndpointUrl) {
+    throw new Error(`"managementEndpointUrl" is missing from the auth file: ${filePath}.`);
+  }
+}
+
+function _foundManagementEndpointUrl(authFileUrl, envUrl) {
+  if (!authFileUrl || (authFileUrl && typeof authFileUrl.valueOf() !== 'string')) {
+    throw new Error('authFileUrl cannot be null or undefined and must be of type string.');
+  }
+
+  if (!envUrl || (envUrl && typeof envUrl.valueOf() !== 'string')) {
+    throw new Error('envUrl cannot be null or undefined and must be of type string.');
+  }
+
+  authFileUrl = authFileUrl.endsWith('/') ? authFileUrl.slice(0, -1) : authFileUrl;  
+  envUrl = envUrl.endsWith('/') ? envUrl.slice(0, -1) : envUrl;
+  return (authFileUrl.toLowerCase() === envUrl.toLowerCase());
+}
+
+/**
+ * Authenticates using the service principal information provided in the auth file. This method will set 
+ * the subscriptionId from the auth file to the user provided environment variable in the options 
+ * parameter or the default AZURE_SUBSCRIPTION_ID.
+ * @param {object} [options] - Optional parameters
+ * @param {string} [options.filePath] - Absolute file path to the auth file. If not provided 
+ * then please set the environment variable AZURE_AUTH_LOCATION.
+ * @param {string} [options.subscriptionEnvVariableName] - The subscriptionId environment variable 
+ * name. Default is 'AZURE_SUBSCRIPTION_ID'.
+ * @param {function} callback - The callback
+ */
+function _withAuthFile(options, callback) {
+  if (!callback && typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  if (!options) options = { filePath: '' };
+  if (!callback) {
+    throw new Error('callback cannot be null or undefined.');
+  }
+
+  let filePath = options.filePath;
```

Comment:
in case it helps, will the following code simplify it?
```javascript
    let filePath = options.filePath || process.env[azureConstants.AZURE_AUTH_LOCATION]` 
    if !filepath:
        return callback(new Error(...))
```